tincr/io/design/rapidsmith.tcl

```diff
@@ -0,0 +1,238 @@
+package provide tincr.io.design 0.0
+package require Tcl 8.5
+package require tincr.cad.design 0.0
+package require tincr.cad.device 0.0
+package require tincr.cad.util 0.0
+
+namespace eval ::tincr:: {
+    namespace export \
+        run_rapidsmith_command
+}
+
+## Executes a command send from a running instance of RapidSmith. The "rs_start" and "rs_end"
+#   are used to communicate with RapidSmith when the command has started or completed.
+#
+proc ::tincr::run_rapidsmith_command { cmd } {
+    puts "rs_start"
+    {*}$cmd
+    puts "rs_end"
+} 
+
+## Tests if the specified bel is a routethrough under a certain configuration. Specifically, 
+#   this function checks that [get_property CONFIG.EQN]="(expected_outpin)=(expected_inpin)"
+#
+# @param bel Name of a bel
+# @param expected_inpin Expected input pin of the routethrough
+# @param expected_outpin Expected output pin of the routethrough
+# @return 1 if the bel matches the expected pattern, 0 otherwise
+proc ::tincr::test_routethrough { bel expected_inpin expected_outpin } {
```

Comment:
So far, all of the functions in this file are for validating RapidSmith design import. The way Vivado communicates with RapidSmith is through the standard out stream, so the results of this function are not returned, they are printed to standard out instead. Also, if you look at line 37, this function returns actual vs. expected values so that meaningful error message can be printed in RapidSmith. This is what differentiates it between the function you referenced.